csharp/test/Apache.Arrow.Adbc.Tests/TestBase.cs

```diff
@@ -497,15 +476,20 @@ private static void ValidateBinaryArrayValue(Func<int, object?> expectedValues,
         /// <summary>
         /// Validates a single values for all results (in the batch).
         /// </summary>
-        /// <param name="value">The value to validate.</param>
         /// <param name="length">The length of the current batch/array.</param>
+        /// <param name="value">The value to validate.</param>
         /// <param name="getter">The getter function to retrieve the actual value.</param>
-        private static void ValidateValue(Func<int, object?> value, int length, Func<int, object?> getter, Int32Array? indexColumn = null)
+        /// <param name="indexColumn"></param>
+        private static void ValidateValue(int length, Func<int, object?> value, Func<int, object?> getter, Int32Array? indexColumn = null, Func<int, bool>? isNullEvaluator = default)
         {
             for (int i = 0; i < length; i++)
             {
                 int valueIndex = indexColumn?.GetValue(i) ?? i;
                 object? expected = value(valueIndex);
+                if (expected == null && isNullEvaluator != null)
+                {
+                    Assert.True(isNullEvaluator(i), "expecting actual value to be null");
```

Comment:
Should the test fail if `expected != null && !isNullEvaluator(i)`?

Reply:
Done. Good catch.